tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/DirectSubmission/LogFactoryGetConfigurationForLoggerInstrumentation.cs

```diff
@@ -44,19 +56,105 @@ public class LogFactoryGetConfigurationForLoggerInstrumentation
+            // and there is no "_isDisposing" field that we can check, so using the ConfigurationTable to keep track
+
+            if (configuration is not null && ConfigurationTable.TryGetValue(configuration, out _))
+            {
+                Log.Debug("We've created a configuration already for NLog - skipping.");
+                return CallTargetState.GetDefault();
+            }
+
+            if (logFactoryProxy is not null && logFactoryProxy.IsDisposing)
+            {
+                // when logging is stopped (e.g., shutdown) the configuration will be set to null
+                // and this instrumentation will get hit again, so to avoid creating a new configuration
+                // and adding the Datadog Direct Submission target we just need to exit here.
+                Log.Debug("NLog LogFactory was being disposed so skipping the logging configuration instrumentations.");
+                return CallTargetState.GetDefault();
+            }
+
+            // we don't want to do logs injection with our custom configuration that we create as there won't be any targets
+            if (tracerManager.Settings.LogsInjectionEnabledInternal && configuration is not null)
+            {
+                LogsInjectionHelper<TTarget>.ConfigureLogsInjection(configuration);
+            }
+
+            // if there isn't a configuration AND we have DirectLogSubmission enabled, create a configuration
+            var setConfigurationRequired = false; // indicate whether we need to set LogFactory.Configuration
+            if (tracerManager.DirectLogSubmission.Settings.IsIntegrationEnabled(IntegrationId.NLog)
+             && configuration is null)
+            {
+                Log.Debug("NLog configuration was null - creating one to allow Direct Log Submission");
+                var loggingConfigurationInstance = Activator.CreateInstance(typeof(TLoggingConfiguration));
+                if (loggingConfigurationInstance is null)
+                {
+                    Log.Warning("Failed to create instance of NLog.Config.LoggingConfiguration Type");
+                    return CallTargetState.GetDefault();
+                }
+
+                setConfigurationRequired = true;
+
+                // update the "ref" configuration passed to us with the new one
+                // Note: we need to call LogFactory.Configuration's setter as well to allow NLog
+                //       to hook everything up correctly for this newly created configuration
+                //       but first we need to create and add the direct logs submission target
+                configuration = (TLoggingConfiguration)loggingConfigurationInstance;
+                Log.Debug("Created custom NLog configuration");
+            }
+
+            if (configuration is not null && logFactoryProxy is null)
+            {
+                // TODO hack not sure what to do here really
+                ConfigurationTable.Add(configuration, new());
```

Comment:
We probably don't need to create an object as we're never using it?
```suggestion
                ConfigurationTable.Add(configuration, null);
```

Reply:
Yeah never need it, I'll swap it to null